core/dbt/parser/source_config.py

```diff
@@ -107,7 +101,7 @@ def update_in_model_config(self, config):
                     )
                 current.update(value)
                 self.in_model_config[key] = current
-            else:  # key in self.ClobberFields
+            else:  # key in self.ClobberFields or self.AdapterSpecificConfigs
```

Comment:
I gave this a tiny bit of thought, and I'm really not sure what an append/extend config could possibly look like for adapter configs. I think "clobber" will be the most common paradigm for new configs, but I do also like the idea of not needing to change this API in the future. I _think_ it will be easy to just add a couple more class attributes like `AdapterSpecificExtendConfigs` as the need arises, and 3rd party adapters should just inherit these attributes from the base adapter. Do you buy that? Or do you think there's a better way of doing this?

Reply:
Yeah, that does sound ok, in the worst case we end up with 6 of these which isn't terrible.

I do think adapter-specific append configs are potentially pretty reasonable - as a lazy example, imagine an adapter where the underlying database supports a concept like tags. It would feel pretty natural for those to append instead of clobbering.